sys/drv/pit.c

```diff
@@ -11,56 +11,76 @@ typedef struct pit_state {
   resource_t *regs;
   resource_t *irq_res;
   timer_t timer;
-  uint16_t period_cntr;      /* period as PIT counter value */
-  uint16_t cntr16_prev_read; /* last read counter value */
-  timercntr_t cntr64;        /* counter value since timer initialization */
+  bool noticed_overflow; /* should we add the counter period */
```

Comment:
The comment here is misleading. If I understand correctly, this field tells us if "we have we noticed and handled overflow" (you use it in that way).